typings/styled-components.d.ts

```diff
@@ -207,9 +90,12 @@ export function injectGlobal(strings: TemplateStringsArray, ...interpolations: S
 
 export const ThemeProvider: ThemeProviderComponent<any>;
 
+export class StyleSheetManager extends React.Component<{ sheet: ServerStyleSheet }, any> { }
+
 export class ServerStyleSheet {
-  collectStyles(children: ReactElement<any>): ReactElement<any>
-  getStyleTags(): string
+  collectStyles(children: ReactElement<any>): StyleSheetManager
+  collectStyles(tree: any): StyleSheetManager;
```

Comment:
It's not really an `any`, so I would argue :) and `ReactNode` would leave just a single simple signature: `collectStyles(tree: React.ReactNode): StyleSheetManager`.
We should be very careful with `any`, it's always a source of many hidden bugs.
My personal opinion is to start with more restrictive types and extend them if needed, other direction is always a breaking change.